src/main/main.c

```diff
@@ -234,7 +234,7 @@ void init(void)
     servoMixerInit(masterConfig.customServoMixer);
 #endif
 
-#ifdef SPEKTRUM_BIND
+#if defined(SERIAL_RX) && defined(SPEKTRUM_BIND)
```

Comment:
Maybe we should take the discussion on build config dependencies and default handling in another thread. We are stalling this PR. I am sorry I started all this....

Revert to your original conditional in main.c , makes sense to only have the "SPEKTRUM_BIND" condition as the code only does "spektrumBind()"  (plus a few other run time conditions).
Add this to CJMCU/target.h:

    #ifdef USE_SERIALRX_SPEKTRUM
    #undef USE_SERIALRX_SPEKTRUM
    #endif
    
    #ifdef SPEKTRUM_BIND
    #undef SPEKTRUM_BIND
    #endif

Then we can merge this and move on...